python/tests/conftest.py

```diff
@@ -231,6 +232,35 @@ def create_version_store(
         shutil.rmtree(tmpdir, ignore_errors=True)
 
 
+def _arctic_library_factory_impl(used, name, arctic_client, library_options) -> Library:
+    """Common logic behind all the factory fixtures"""
+    name = name or default_name
+    if name == "_unique_":
+        name = name + str(len(used))
+    assert name not in used, f"{name} is already in use"
+    arctic_client.create_library(name, library_options)
+    out = arctic_client[name]
+    used[name] = out
+    return out
+
+
+@pytest.fixture(scope="function")
+def library_factory(arctic_client, lib_name):
+    used: Dict[str, Library] = {}
+
+    def create_library(
+        library_options: Optional[LibraryOptions] = None,
+    ) -> Library:
+        return _arctic_library_factory_impl(used, lib_name, arctic_client, library_options)
+
+    return create_library
+
+
+@pytest.fixture(scope="function")
+def library_factory_small_segment_size(library_factory):
+    return library_factory(LibraryOptions(rows_per_segment=10))
```

Comment:
The idea is to not have fixtures like this. So `test_write_batch` should use the `library_factory` fixture directly, and pass in appropriate `LibraryOptions` to create a library with the desired configuration.

Reply:
agree